README.md

```diff
@@ -1 +1,649 @@
+
+# 5 Disconnection
+
+```
+Requestor                  Responder
+    ⋮                          ⋮
+    │                          │
+    │           FIN            │ (7)
+    │◄────────────────────────►│
+    │                          │
+    ▀                          ▀ 
+```
+
+Graceful disconnection from an AWAKE attempt MAY be broadcast at any time during an AWAKE session, including to cancel the AWAKE handshake attempt. This payload MUST be the standard [AWAKE message type](#4-secure-session), SHOULD NOT contain any other keys.
+
+### 5.1 Encrypted Field Keys
+
+The disconnection message MUST include an `awake/fin` key with `disconnect` for its value. It MAY include additional fields.
+
+| Field       | Value        | Description             | Required |
+| ----------- | ------------ | ----------------------- | -------- |
+| `awake/fin` | `disconnect` | Disconnection directive | Yes      |
+
+``` javascript
+// JSON encoded
+{
+  ...,
+  "awake/fin": "disconnect"
+}
+```
+
+# 6 Errors
+
+### 6.1 Cleartext Envelope
+
+All errors MUST use the generic [AWAKE message payload](#4-secure-session), and include the error information in the encrypted payload. It MUST use the latest ECDH keys.
+
+## 6.2 Unknown Challenge Error
+
+| Field         | Value               | Description                         | Required |
+| ------------- | ------------------- | ----------------------------------- | -------- |
+| `awake/error` | `unknown-challenge` | Unknown challenge type              | Yes      |
+| `awake/mid`   |                     | Message ID that generated the error | Yes      |
+
+``` javascript
+// JSON encoded
+{
+  "awake/error": "unknown-challenge",
+  "awake/mid": offendingMessageId
+}
+```
+
+# 7 Prior Art
+
+## 7.1 Mutual TLS (mTLS)
+
+[mTLS](https://www.rfc-editor.org/rfc/rfc8705.html) is the best-known mutual authentication protocol. In many ways, AWAKE is mTLS with trusted rooted in UCAN and a self-signed capabilities model.
+
+## 7.2 IKEv2
+
+The [Internet Key Exchange (IKE) Protocol](https://datatracker.ietf.org/doc/html/rfc7296) is typically (but not exclusively) used as part of [IPsec](https://en.wikipedia.org/wiki/IPsec). IKE generally uses certificate authorities (CAs). IKE requires that X.509 be supported.
+
+IKE shares many commonalities with AWAKE, including making available of the same cryptographic algorithms (e.g. P-256).
+
+## 7.3 WireGuard
+
+[WireGuard](https://www.wireguard.com/) is a VPN protocol that is widely deployed via the Linux kernel, and has since been ported to many other systems. It is UDP-based and aimed at raw performance and security. Being so level, is unconstrained in which cryptographic primitives is uses (i.e. Curve25519).
```

Comment:
> Being so level, is unconstrained in which cryptographic primitives is uses (i.e. Curve25519).

Should this sentence open with "Being low-level, it is..."? May be something else, but I'm guessing that might be it because it's in the kernel.

Reply:
Indeed! Fixed 🙏